### ISSUE TITLE
Reconcile Calico's IPPool Object

### ISSUE THREAD

**Issue by pschichtel** [k0sproject/k0s#4061]
### Is your feature request related to a problem? Please describe.

I've changed my calico CNI configuration from `overlay: Always` to `overlay: CrossSubnet`. I'm aware that k0s does not currently support changing CNI providers, but I figured that I can just change the k0s config, see how for that gets me and cleanup the rest manually, given that I have prior experience with self-managing calico.

The state as of now seems to be that k0s reconciles the calico deployment partially based on the config change. I noticed that the daemonset gets updated correctly (it carries `CALICO_IPV4POOL_IPIP` with the `overlay` value) and possibly other objects in the cluster, however it does not seem to update the IPPool configuration.

### Describe the solution you would like

It would be nice if k0s would reconcile calico's configuration objects like IPPool as well.

### Describe alternatives you've considered

Alternatively it would be neat to have some kind of warning message or other signal, that the CNI configuration changed, but not all changes have been applied to the cluster.

**Comment by pschichtel** [comment]
I'm fully aware that I'm sailing in dangerous waters here, so I'm perfectly ok with this getting rejected for being out of scope or not worth the risk of breaking something else in the reconciliation process.

**Comment by twz123** [comment]
K0s [actually](https://github.com/k0sproject/k0s/blob/8c29491e3aa4c6f1b37cd9814da32f225922ca05/pkg/component/controller/calico.go#L196) [reconciles](https://github.com/k0sproject/k0s/blob/8c29491e3aa4c6f1b37cd9814da32f225922ca05/static/manifests/calico/DaemonSet/calico-node.yaml#L206-L210) the pod CIDR in the Calico manifests, but only in a way that's picked up on first installation. The subsequent config changes are ignored by Calico if the IPPool exists.

Changing this would probably mean to create the IPPool manually, instead of delegating this to Calico.

OTOH, is it safe to change the pod CIDR in a running cluster? K0s at least refuses to do it for the service CIDR and cluster domain...

**Comment by pschichtel** [comment]
Yeah changing the networks has probably a bigger impact, I've never attempted that, but changing the overlay mode was picked up immediately by calico (I watched the routes change)
